src/artifacts_poetry_plugin/plugin.py

```diff
@@ -9,143 +9,234 @@
-
-
-class ArtifactsDeploy(Command):
+            raise RuntimeError(err_msg)
+        if username is None or password is None:
+            raise RuntimeError(err_msg)
+        return RepoData(url, username, password)
+
+    def get_poetry_packages(self, poetry) -> List[Package]:
+        """Gets packages from the given poetry.lock file"""
+        locker = poetry.locker
+        repository = locker.locked_repository()
+        return repository.packages
+
+    def get_cached_wheel_files(self, cache_dir):
+        """Gets all wheel file locations for the given cache directory"""
+        wheel_files = {}
+        for root, dirs, files in os.walk(cache_dir):
+            for file in files:
+                if file.endswith(".whl") or file.endswith(".tar.gz"):
+                    wheel_files[file] = Path(os.path.join(root, file))
+        return wheel_files
+
+    def get_deploy_non_deploy_packages(self, packages: List[Package], wheel_files):
+        """Adds the wheel file locations for each package's files"""
+        non_deployable_packages = []
+        deployable_packages = []
+        for package in packages:
+            file_found = False
+            for file in package.files:
+                if file["file"] in wheel_files:
+                    file_found = True
+                    file["url"] = wheel_files[file["file"]]
+            if file_found:
+                deployable_packages.append(package)
+            else:
+                non_deployable_packages.append(package)
+        return deployable_packages, non_deployable_packages
+
+    def upload_packages(self, poetry_files, url, username, password, io):
+        deploy_set = set()
+        for poetry in poetry_files:
+            uploader = Uploader(poetry, io)
+            uploader.auth(username=username, password=password)
+            session = uploader.make_session()
+            for package in poetry.deployable_packages:
+                uploader._package = package
+                for file in package.files:
+                    if "url" in file and file["url"] not in deploy_set:
+                        deploy_set.add(file["url"])
+                        uploader._upload_file(
+                            session, url, file=file["url"], skip_existing=True
+                        )
+
+    def get_project_package(self, poetry):
+        """Adds the generated package to the list of packages with its associated wheel file"""
+        project_package_path = os.path.join(poetry.root_dir, "dist")
+        project_package = poetry.package
+        for root, dirs, files in os.walk(project_package_path):
+            for file in files:
+                if file.endswith(".whl") or file.endswith(".tar.gz"):
+                    file_data = {"file": file, "url": Path(os.path.join(root, file))}
+                    project_package.files.append(file_data)
+        return project_package
+
+    def get_all_poetry(self, path, downstream):
```

Comment:
docstrings might be helpful here, esp because it's not clear at first what the `downstream` conveys. It seem like if downstream is false, then we need to instantiate a poetry object and set the root dir?

Reply:
Added one to describe its use.